src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -84,8 +85,8 @@ public sealed partial class TextLoader
                         $"Valid characters are 0-9, *, - and ~");
 
                 var name = mappingAttr.Name ?? field.Name;
-                if (name.Any(c => !Char.IsLetterOrDigit(c)))
-                    throw Contracts.Except($"{name} is not alphanumeric.");
+                if (!new CSharpCodeProvider().IsValidIdentifier(name))
+                    throw Contracts.Except($"{name} is not a valid identifier.");
 
```

Comment:
High level question -- I *believe* this is now using the new-ish API for v0.2 that utilizes the properties directly, and does not attempt to work through the goofy command line parsing that, IIRC, made this check necessary. So why do we still have it *at all*?